Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1292,18 +1335,18 @@ private extension Purchases {
     }
 
     func updateAllCachesIfNeeded() {
-        systemInfo.isApplicationBackgrounded { isAppBackgrounded in
-            self.customerInfoManager.fetchAndCacheCustomerInfoIfStale(appUserID: self.appUserID,
-                                                                      isAppBackgrounded: isAppBackgrounded,
-                                                                      completion: nil)
-            guard self.deviceCache.isOfferingsCacheStale(isAppBackgrounded: isAppBackgrounded) else {
-                return
+        self.systemInfo.isApplicationBackgrounded { isAppBackgrounded in
+            if !self.systemInfo.dangerousSettings.minimalImplementationOnly {
+                self.customerInfoManager.fetchAndCacheCustomerInfoIfStale(appUserID: self.appUserID,
+                                                                          isAppBackgrounded: isAppBackgrounded,
+                                                                          completion: nil)
             }
 
-            Logger.debug("Offerings cache is stale, updating caches")
-            self.offeringsManager.updateOfferingsCache(appUserID: self.appUserID,
-                                                       isAppBackgrounded: isAppBackgrounded,
-                                                       completion: nil)
+            if self.deviceCache.isOfferingsCacheStale(isAppBackgrounded: isAppBackgrounded) {
+                self.offeringsManager.updateOfferingsCache(appUserID: self.appUserID,
+                                                           isAppBackgrounded: isAppBackgrounded,
+                                                           completion: nil)
+            }
```

Comment:
this just reorders the methods so that I could do an early return, but I could roll back the ordering and use an if instead